### ISSUE TITLE
Investigate July Update Issues

### ISSUE THREAD

**Issue by dpaulson45** [dpaulson45/HealthChecker#237]
Issue appears to be that newer CUs overwrite the KB, which I would expect but we weren't seeing it at the time this issue was first detected. This is why we just looked for the KB ID in general. Need to investigate this to see if we need to change this. 

July Update detected: Error --- Problem KB4338824 detected without the fix KB4345424. This can cause odd issues to occur on the system. See https://techcommunity.microsoft.com/t5/Exchange-Team-Blog/Issue-with-July-Updates-for-Windows-on-an-Exchange-Server/ba-p/608057

2018-07 Update for Windows Server 2012 R2 for x64-based Systems (KB4345424)
Last Modified: 7/16/2018
Size: 39.0 MB

Has been Superseded by Quite a list

This update has been replaced by the following updates:
2018-08 Security Monthly Quality Rollup for Windows Server 2012 R2 for x64-based Systems (KB4343898)
2018-09 Security Monthly Quality Rollup for Windows Server 2012 R2 for x64-based Systems (KB4457129)
2018-10 Security Monthly Quality Rollup for Windows Server 2012 R2 for x64-based Systems (KB4462926)
2018-11 Security Monthly Quality Rollup for Windows Server 2012 R2 for x64-based Systems (KB4467697)
2018-12 Security Monthly Quality Rollup for Windows Server 2012 R2 for x64-based Systems (KB4471320)
2019-01 Security Monthly Quality Rollup for Windows Server 2012 R2 for x64-based Systems (KB4480963)
2019-02 Security Monthly Quality Rollup for Windows Server 2012 R2 for x64-based Systems (KB4487000)
2019-03 Security Monthly Quality Rollup for Windows Server 2012 R2 for x64-based Systems (KB4489881)
2019-04 Security Monthly Quality Rollup for Windows Server 2012 R2 for x64-based Systems (KB4493446)
2019-05 Security Monthly Quality Rollup for Windows Server 2012 R2 for x64-based Systems (KB4499151)
2019-06 Security Monthly Quality Rollup for Windows Server 2012 R2 for x64-based Systems (KB4503276)
2019-07 Security Monthly Quality Rollup for Windows Server 2012 R2 for x64-based Systems (KB4507448)
2019-08 Security Monthly Quality Rollup for Windows Server 2012 R2 for x64-based Systems (KB4512488)
2019-09 Security Monthly Quality Rollup for Windows Server 2012 R2 for x64-based Systems (KB4516067)
2019-10 Security Monthly Quality Rollup for Windows Server 2012 R2 for x64-based Systems (KB4520005)
2019-10 Security Monthly Quality Rollup for Windows Server 2012 R2 for x64-based Systems (KB4524156)
2019-11 Security Monthly Quality Rollup for Windows Server 2012 R2 for x64-based Systems (KB4525243)

I Had a Customer that ran the healthchecker and got the error but he has KB4520005 installed.

Original issue: #69 
Commit: bce37167cee5e58eab02a724e9bf8e6b9866563e

**Comment by dpaulson45** [comment]
@lusassl-msft are you able to look into this?

**Comment by lusassl-msft** [comment]
I will check this. Please assign to me.

**Comment by lusassl-msft** [comment]
I've looked into this. It seems that the required updates mentioned in the blog post (https://techcommunity.microsoft.com/t5/Exchange-Team-Blog/Issue-with-July-Updates-for-Windows-on-an-Exchange-Server/ba-p/608057) have been replaced with newer Quality Rollups (Security Monthly Quality Rollup for Windows Server 20XX). If a new Security Monthly Quality Rollup has been released, it'll be listed to replace the mentioned update. 

What we can do here is to check if an problematic update is detected and if we don't detect the corresponding update which addresses the issue, we may check if we detect any Security Monthly Quality update with a newer release date than the initial _update which must be applied_. Doing it this way helps us to not add each update which supersedes the initial one month by month. What do you think?

**Comment by lusassl-msft** [comment]
Hm not as easy as I thought (because of different system language it's hard to parse the return accurate).

**Comment by dpaulson45** [comment]
closing out as wont fix as in Dev3.0.0 we decided to not include this any longer.
